src/source/geojson_source.js

```diff
@@ -190,7 +190,7 @@ class GeoJSONSource extends Evented implements Source {
             type: this.type,
             uid: tile.uid,
             coord: tile.coord,
-            zoom: tile.coord.z,
+            zoom: this.map.transform.zoom,
```

Comment:
@asheemmamoowala I'd love to avoid another point of dependency of sources on the `map` object (see also https://github.com/mapbox/mapbox-gl-js/issues/3350).

Would an alternative to using the current map's zoom level be to keep using `tile.coord.z`, but change the test to something like `if (this.zoom < floor(layer.minzoom)) continue;`?

If I'm understanding this right, we want:

layer minzoom | tile zoom | tile needed?
---------------|----------|-------------
4|3 (and below)|no
4|4 (and above)|yes
4.5|4 (and above)|yes
4.5|3 (and below)|no